### PR TITLE
Add section on closing issues to core dev guide

### PR DESCRIPTION
**Pull request by jni**
We should aim as core developers to ensure that community members and contributors feel welcome even after their issues or pull requests are closed. I hope this section of the guide helps that goal.

Comments/suggestions for improvement are welcome!

### CODE REVIEW

File: doc/source/core_developer.rst [hunk 1]
```diff
@@ -122,6 +122,49 @@ While we collectively "own" any patches (and bugs!) that become part
...
+among others. Similarly, pull requests sometimes need to be closed without
+merging, because:
+
+- the pull request implements a niche feature that we consider not worth the
+  added maintenance burden;
+- the pull request implements a useful feature, but requires significant
+  effort to bring up to scikit-image's standards, and the original
+  contributor has moved on, and no other developer can be found to make the
+  necessary changes; or
+- the pull request makes changes that do not align with our values, such as
```
Comment (hmaarrfk) [inline doc/source/core_developer.rst]: Should you remove everything ater the "does not align with our values"

File: doc/source/core_developer.rst [hunk 2]
```diff
@@ -122,6 +122,49 @@ While we collectively "own" any patches (and bugs!) that become part
...
+  increasing the code complexity of a function significantly to implement a
+  marginal speedup,
+
+among others.
+
+All these are valid reasons for closing, but we must be wary not to alienate
```
Comment (hmaarrfk) [inline doc/source/core_developer.rst]:
```suggestion
All these may be valid reasons for closing, but we must be wary not to alienate
```

File: doc/source/core_developer.rst [hunk 3]
```diff
@@ -122,6 +122,49 @@ While we collectively "own" any patches (and bugs!) that become part
...
+contributors by closing an issue or pull request without an explanation. When
+closing, your message should:
+
+- explain clearly how the decision was made to close. This is particularly
+  important when the decision was made in a community meeting, which does not
```
Comment (hmaarrfk) [inline doc/source/core_developer.rst]: This is funny isn't it. Community meeting are supposed to increase visibility to the project, but maybe have the opposite effect?

File: doc/source/core_developer.rst (hunk 1 above)
Comment (emmanuelle) [reply doc/source/core_developer.rst]: I would keep the example, it's a good illustration of the kind of things which we do not want.

**Comment by emmanuelle** [comment]
Thank you @jni ! Could you please make the modification suggested by @hmaarrfk (are --> may be) if you're happy with it, and then we merge this one?

**Comment by jni** [comment]
@emmanuelle done! Thanks all for the reviews!

File: doc/source/core_developer.rst (hunk 3 above)
Comment (jni) [reply doc/source/core_developer.rst]: They only have the opposite effect if we are bad at communicating their outcomes afterwards. =) But, yes!

**Pull request merged**